code/modules/paperwork/contract.dm

```diff
@@ -325,8 +325,12 @@
 /obj/item/paper/contract/infernal/knowledge/fulfillContract(mob/living/carbon/human/user = target.current, blood = 0)
 	if(!istype(user) || !user.mind)
 		return -1
-	user.dna.add_mutation(XRAY)
-	user.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/view_range(null))
+	
+	for(var/V in user.quirk_list)
+		var/datum/quirk/T = V
+		if (T.value<0)
+			T.Destroy()	
+	user.fully_heal(FALSE)
```

Comment:
I don't see any reason for anyone taking this, since if they are alive and can sign they can already heal themselves and if they are dead there is a revive contract that does the exact same thing